src/engine/Collision/Collider.ts

```diff
@@ -0,0 +1,154 @@
+import { Color } from '../Drawing/Color';
+import * as DrawUtil from '../Util/DrawUtil';
+import { Eventable } from '../Interfaces/Index';
+import { GameEvent, PostCollisionEvent, PreCollisionEvent, CollisionStartEvent, CollisionEndEvent } from '../Events';
+import { Actor } from '../Actor';
+import { Body } from './Body';
+import { CollisionGeometry } from './CollisionGeometry';
+import { Vector } from '../Algebra';
+import { Physics, CollisionResolutionStrategy } from '../Physics';
+import { BoundingBox } from './BoundingBox';
+import { ConvexPolygon } from './ConvexPolygon';
+import { CollisionType } from './CollisionType';
+
+function isCollider(x: Actor | Collider): x is Collider {
+  return !!x && x instanceof Collider;
+}
+
+// Describes material properties like shape, bounds, friction of the physics object
+export class Collider implements Eventable {
+  private _collisionArea: CollisionGeometry;
+
+  constructor(private _actor: Actor, private _body: Body) {}
+
+  emit(eventName: string, event?: GameEvent<any>): void {
```

Comment:
I'm not super pleased with this, maybe we don't this long term? Perhaps come up with some automatic bubbling strategy?

Reply:
Put these events into a array/dictionary and lookup the type and redispatch